src/Microsoft.AspNetCore.Razor/Compilation/TagHelpers/TagHelperRequiredAttributeDescriptor.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.AspNetCore.Razor.Compilation.TagHelpers
+{
+    public class TagHelperRequiredAttributeDescriptor
+    {
+        public string Name { get; set; }
+
+        public string Value { get; set; }
+
+        public char Selector { get; set; }
+
+        public bool IsCSSSelector { get; set; }
+
+        public bool Matches(string attributeName, string attributeValue)
+        {
+            if (IsCSSSelector)
+            {
+                var nameMatches = string.Equals(Name, attributeName, StringComparison.OrdinalIgnoreCase);
+
+                if (!nameMatches)
+                {
+                    return false;
+                }
+
+                var valueMatches = false;
+
+                switch (Selector)
+                {
+                    case '^': // Value starts with
```

Comment:
This looks confusing and a bit bizarre. Probably much better to make `Selector` an `enum` identifying the comparison operator. E.g. the "starts with" operator is `^=`. (Could even use the CSS names for the operators -- `PREFIXMATCH`, `SUFFIXMATCH` and `EQUALS`. Well, except the grammar doesn't name `'='`.)

Also not positive about the name `Selector` since "attribute selector" refers to the whole `[ name ]` or `[ name <op> value ]`.